src/devices/bus/nscsi/tape.h

```diff
@@ -0,0 +1,115 @@
+// license:BSD-3-Clause
+// copyright-holders:Mietek Bak
+
+#ifndef MAME_BUS_NSCSI_TAPE_H
+#define MAME_BUS_NSCSI_TAPE_H
+
+#pragma once
+
+#include "machine/nscsi_bus.h"
+#include "imagedev/simh_tape_image.h"
+
+DECLARE_DEVICE_TYPE(NSCSI_TAPE, nscsi_tape_device);
+
+//////////////////////////////////////////////////////////////////////////////
+
+class nscsi_tape_device : public nscsi_full_device
+{
+public:
+	// construction
+	nscsi_tape_device(const machine_config &config, const char *tag, device_t *owner, uint32_t clock = 0);
+
+protected:
+	// basic state
+	int m_fixed_block_len; // fixed-length block length
+	required_device<simh_tape_image_device> m_image; // tape image
+	int m_sequence_counter; // tape image identifier
+	bool m_tape_changed; // should we report unit attention next time we receive command
+
+	// state for READ and WRITE
+	uint8_t m_rw_buf[32768]; // read/write buffer; TODO: adjust size or make it configurable, once we encounter tape images that use larger block lengths
```

Comment:
MAME prefers a std::unique_ptr<uint8_t> for this kind of buffer.  That way you can dynamically allocate it to match the block size as needed.  (See m_vram in src/mame/apple/dafb.cpp for an example).